### ISSUE TITLE
ffmpeg not working in Linux

### ISSUE THREAD

**Issue by GoogleCodeExporter** [0xfab-ri/universal-media-server#32]
```
What steps will reproduce the problem?
1. Set FFMPEG to top priority encoder
2. Attempt to play a video

What is the expected output? What do you see instead?
INFO  2013-01-01 05:35:06.602 [New I/O server worker #3-2] Starting 
transcode/remux of Alien_3_(Special_Edition)_(1992).mp4
DEBUG 2013-01-01 05:35:06.635 [New I/O server worker #3-2] Caught exception
java.lang.NullPointerException: null
    at net.pms.encoders.MEncoderVideo.launchTranscode(MEncoderVideo.java:1349) ~[ums.jar:2.2.1]
    at net.pms.dlna.DLNAResource.getInputStream(DLNAResource.java:1747) ~[ums.jar:2.2.1]
    at net.pms.network.RequestV2.answer(RequestV2.java:326) ~[ums.jar:2.2.1]
    at net.pms.network.RequestHandlerV2.writeResponse(RequestHandlerV2.java:313) ~[ums.jar:2.2.1]
    at net.pms.network.RequestHandlerV2.messageReceived(RequestHandlerV2.java:270) ~[ums.jar:2.2.1]
    at org.jboss.netty.handler.stream.ChunkedWriteHandler.handleUpstream(ChunkedWriteHandler.java:149) ~[ums.jar:2.2.1]
    at org.jboss.netty.handler.codec.http.HttpChunkAggregator.messageReceived(HttpChunkAggregator.java:116) ~[ums.jar:2.2.1]
    at org.jboss.netty.channel.Channels.fireMessageReceived(Channels.java:302) ~[ums.jar:2.2.1]
    at org.jboss.netty.handler.codec.replay.ReplayingDecoder.unfoldAndFireMessageReceived(ReplayingDecoder.java:527) ~[ums.jar:2.2.1]
    at org.jboss.netty.handler.codec.replay.ReplayingDecoder.callDecode(ReplayingDecoder.java:506) ~[ums.jar:2.2.1]
    at org.jboss.netty.handler.codec.replay.ReplayingDecoder.messageReceived(ReplayingDecoder.java:443) ~[ums.jar:2.2.1]
    at org.jboss.netty.channel.Channels.fireMessageReceived(Channels.java:274) ~[ums.jar:2.2.1]
    at org.jboss.netty.channel.Channels.fireMessageReceived(Channels.java:261) ~[ums.jar:2.2.1]
    at org.jboss.netty.channel.socket.nio.NioWorker.read(NioWorker.java:351) ~[ums.jar:2.2.1]
    at org.jboss.netty.channel.socket.nio.NioWorker.processSelectedKeys(NioWorker.java:282) ~[ums.jar:2.2.1]
    at org.jboss.netty.channel.socket.nio.NioWorker.run(NioWorker.java:202) ~[ums.jar:2.2.1]
    at java.util.concurrent.ThreadPoolExecutor.runWorker(ThreadPoolExecutor.java:1110) [na:1.7.0_09]
    at java.util.concurrent.ThreadPoolExecutor$Worker.run(ThreadPoolExecutor.java:603) [na:1.7.0_09]
    at java.lang.Thread.run(Thread.java:722) [na:1.7.0_09]
INFO  2013-01-01 05:35:06.822 [New I/O server worker #3-3] Starting 
transcode/remux of Alien_3_(Special_Edition)_(1992).mp4
DEBUG 2013-01-01 05:35:06.850 [New I/O server worker #3-3] Caught exception
java.lang.NullPointerException: null
    at net.pms.encoders.MEncoderVideo.launchTranscode(MEncoderVideo.java:1349) ~[ums.jar:2.2.1]
    at net.pms.dlna.DLNAResource.getInputStream(DLNAResource.java:1747) ~[ums.jar:2.2.1]
    at net.pms.network.RequestV2.answer(RequestV2.java:326) ~[ums.jar:2.2.1]
    at net.pms.network.RequestHandlerV2.writeResponse(RequestHandlerV2.java:313) ~[ums.jar:2.2.1]
    at net.pms.network.RequestHandlerV2.messageReceived(RequestHandlerV2.java:270) ~[ums.jar:2.2.1]
    at org.jboss.netty.handler.stream.ChunkedWriteHandler.handleUpstream(ChunkedWriteHandler.java:149) ~[ums.jar:2.2.1]
    at org.jboss.netty.handler.codec.http.HttpChunkAggregator.messageReceived(HttpChunkAggregator.java:116) ~[ums.jar:2.2.1]
    at org.jboss.netty.channel.Channels.fireMessageReceived(Channels.java:302) ~[ums.jar:2.2.1]
    at org.jboss.netty.handler.codec.replay.ReplayingDecoder.unfoldAndFireMessageReceived(ReplayingDecoder.java:527) ~[ums.jar:2.2.1]
    at org.jboss.netty.handler.codec.replay.ReplayingDecoder.callDecode(ReplayingDecoder.java:506) ~[ums.jar:2.2.1]
    at org.jboss.netty.handler.codec.replay.ReplayingDecoder.messageReceived(ReplayingDecoder.java:443) ~[ums.jar:2.2.1]
    at org.jboss.netty.channel.Channels.fireMessageReceived(Channels.java:274) ~[ums.jar:2.2.1]
    at org.jboss.netty.channel.Channels.fireMessageReceived(Channels.java:261) ~[ums.jar:2.2.1]
    at org.jboss.netty.channel.socket.nio.NioWorker.read(NioWorker.java:351) ~[ums.jar:2.2.1]
    at org.jboss.netty.channel.socket.nio.NioWorker.processSelectedKeys(NioWorker.java:282) ~[ums.jar:2.2.1]
    at org.jboss.netty.channel.socket.nio.NioWorker.run(NioWorker.java:202) ~[ums.jar:2.2.1]
    at java.util.concurrent.ThreadPoolExecutor.runWorker(ThreadPoolExecutor.java:1110) [na:1.7.0_09]
    at java.util.concurrent.ThreadPoolExecutor$Worker.run(ThreadPoolExecutor.java:603) [na:1.7.0_09]
    at java.lang.Thread.run(Thread.java:722) [na:1.7.0_09]


What version of the product are you using? On what operating system?

UMS 2.2.1 on Ubuntu 12.10 AMD64
ffmpeg version 0.8.4-6:0.8.4-0ubuntu0.12.10.1
```

Original issue reported on code.google.com by `anarchoc...@gmail.com` on 1 Jan 2013 at 10:38

**Comment by GoogleCodeExporter** [comment]
```
Is this still happening with 2.2.5?
```

Original comment by `subjunk` on 19 Jan 2013 at 10:16

**Comment by GoogleCodeExporter** [comment]
```
I got the following with 2.2.6. Looks like incompatible version of ffmpeg. 
While its more work I'd recommend providing statically linked versions of the 
tools needed so this kind of thing is less likely to happen.

INFO 2013-01-24 09:21:34.683 [New I/O server worker #1-2] Starting 
transcode/remux of 2001_-_A_Space_Odyssey.mkv
DEBUG 2013-01-24 09:21:34.685 [New I/O server worker #1-2] Custom ffmpeg output 
options: -g 5 -q:v 1 -qmin 2
DEBUG 2013-01-24 09:21:34.685 [ffmpeg-416] Starting ffmpeg -y -loglevel warning 
-threads 4 -i /media/video/Movies/2001_-
_A_Space_Odyssey.mkv -threads 4 -bufsize 1835000 -maxrate 54000000 -ab 640k -g 
5 -q:v 1 -qmin 2 -c:a copy -f vob -c:v mp
eg2video pipe:
DEBUG 2013-01-24 09:21:34.695 [ffmpeg-416] Attaching thread: ffmpeg-416
DEBUG 2013-01-24 09:21:34.695 [ffmpeg-416] Unix process ID (ffmpeg): 23235
DEBUG 2013-01-24 09:21:34.700 [Thread-1635] *** THIS PROGRAM IS DEPRECATED ***
DEBUG 2013-01-24 09:21:34.700 [Thread-1635] This program is only provided for 
compatibility and will be removed in a future release. Please use avconv 
instead.
DEBUG 2013-01-24 09:21:34.712 [New I/O server worker #1-3] Caught exception
java.lang.StringIndexOutOfBoundsException: String index out of range: -1
at java.lang.String.substring(String.java:1911) ~[na:1.7.0_09]
at net.pms.network.RequestV2.answer(RequestV2.java:263) ~[ums.jar:2.2.6]
at net.pms.network.RequestHandlerV2.writeResponse(RequestHandlerV2.java:313) 
~[ums.jar:2.2.6]
at net.pms.network.RequestHandlerV2.messageReceived(RequestHandlerV2.java:270) 
~[ums.jar:2.2.6]
at 
org.jboss.netty.handler.stream.ChunkedWriteHandler.handleUpstream(ChunkedWriteHa
ndler.java:149) ~[ums.jar:2.2
.6]
at 
org.jboss.netty.handler.codec.http.HttpChunkAggregator.messageReceived(HttpChunk
Aggregator.java:116) ~[ums.ja
r:2.2.6]
at org.jboss.netty.channel.Channels.fireMessageReceived(Channels.java:302) 
~[ums.jar:2.2.6]
at 
org.jboss.netty.handler.codec.replay.ReplayingDecoder.unfoldAndFireMessageReceiv
ed(ReplayingDecoder.java:527)
~[ums.jar:2.2.6]
at 
org.jboss.netty.handler.codec.replay.ReplayingDecoder.callDecode(ReplayingDecode
r.java:506) ~[ums.jar:2.2.6]
at 
org.jboss.netty.handler.codec.replay.ReplayingDecoder.messageReceived(ReplayingD
ecoder.java:443) ~[ums.jar:2.
2.6]
at org.jboss.netty.channel.Channels.fireMessageReceived(Channels.java:274) 
~[ums.jar:2.2.6]
at org.jboss.netty.channel.Channels.fireMessageReceived(Channels.java:261) 
~[ums.jar:2.2.6]
at org.jboss.netty.channel.socket.nio.NioWorker.read(NioWorker.java:351) 
~[ums.jar:2.2.6]
at 
org.jboss.netty.channel.socket.nio.NioWorker.processSelectedKeys(NioWorker.java:
282) ~[ums.jar:2.2.6]
at org.jboss.netty.channel.socket.nio.NioWorker.run(NioWorker.java:202) 
~[ums.jar:2.2.6]
at 
java.util.concurrent.ThreadPoolExecutor.runWorker(ThreadPoolExecutor.java:1110) 
[na:1.7.0_09]
at 
java.util.concurrent.ThreadPoolExecutor$Worker.run(ThreadPoolExecutor.java:603) 
[na:1.7.0_09]
at java.lang.Thread.run(Thread.java:722) [na:1.7.0_09]
DEBUG 2013-01-24 09:21:34.734 [Thread-1635] [matroska,webm @ 0x6c69e0] 
Estimating duration from bitrate, this may be ina
ccurate
DEBUG 2013-01-24 09:21:34.735 [Thread-1635]
DEBUG 2013-01-24 09:21:34.735 [Thread-1635] Seems stream 0 codec frame rate 
differs from container frame rate: 47.95 (50
00000/104271) -> 23.98 (24000/1001)
DEBUG 2013-01-24 09:21:34.735 [Thread-1635] Unrecognized option 'q:v'
DEBUG 2013-01-24 09:21:34.735 [Thread-1635] Failed to set value '1' for option 
'q:v'
INFO 2013-01-24 09:21:34.736 [ffmpeg-416] Process ffmpeg has a return code of 
1! Maybe an error occurred... check the log file
```

Original comment by `anarchoc...@gmail.com` on 24 Jan 2013 at 2:25

**Comment by GoogleCodeExporter** [comment]
```
I agree and it has been our plan for a long time to include binaries for all 
operating systems. Hopefully we will do that soon :)
```

Original comment by `subjunk` on 24 Jan 2013 at 8:30

**Comment by GoogleCodeExporter** [comment]
```
At the one side I would like to see statically linked binaries of mencoder, 
mplayer, ffmpeg, x.264 and so on.

At the other side I remember sleepless nights to find and build performant 
tools for my debian GNU/linux testing x64. A lot of measurement and choices 
will be made to decide for or against mplayer or mplayer2, mencoder or 
mencoder-mt, ffmpeg including mutlithreading support or not.

I believe it should be not your work to support a few binaries for a lot of 
opaerating systems in i686 and x86_64 (or other platforms).

Nice idea, but hard work.
```

Original comment by `ronnysef...@gmail.com` on 28 Jan 2013 at 10:43

**Comment by GoogleCodeExporter** [comment]
```
Static FFmpeg, MPlayer and MEncoder will be included in 2.4.0 :)
```

Original comment by `subjunk` on 9 Feb 2013 at 9:34
- Changed state: **Fixed**
